### PR TITLE
[Enhancement](metadata) show full columns support for external tables

### LINKED ISSUE #25899: [Feature] SHOW FULL COLUMNS Query Support for External Hive Catalog
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/doris/issues?q=is%3Aissue) and found no similar issues.

### Description

I would like to request the addition of support for the "SHOW FULL COLUMNS" query in the External Hive catalog in Doris. This feature would be highly beneficial for our use case and would improve the integration with external Hive metadata.

### Use case

I am trying to use Apache Calcite's algebra builder with Apache Doris, and I have a issue related to query compatibility.
My goal is to use the algebra builder/RelBuilder(https://calcite.apache.org/docs/algebra.html). Calcite will gather all the necessary column details for creating RelBuilder by calling the getColumns() method in MySQL connector which executes SHOW FULL COLUMNS query eg: "SHOW FULL COLUMNS FROM instant_cdr from cvm". But SHOW FULL COLUMNS isn't supported for the External catalog(throwing error External catalog 'hive' is not allowed in 'ShowColumnStmt').

### Related issues

_No response_

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by Nitin-Kashyap**
## Proposed changes

Support for the "SHOW FULL COLUMNS" query in the External Hive catalog in Doris.

Issue Number: close #25899

Add suport for SHOW FULL COLUMNS for external tables.

## Further comments
Additionally added check for column permissions in SHOW FULL COLUMNS path; this was present in Describe, but missing in SHOW FULL COLUMNS.

### CODE REVIEW

**Comment by Nitin-Kashyap** [comment]
run buildall

**Comment by Nitin-Kashyap** [comment]
run buldall

File: fe/fe-core/src/main/java/org/apache/doris/analysis/ShowColumnStmt.java [hunk 1]
```diff
@@ -98,8 +106,14 @@ public void analyze(Analyzer analyzer) throws AnalysisException {
             tableName.setDb(db);
         }
         tableName.analyze(analyzer);
-        // disallow external catalog
-        Util.prohibitExternalCatalog(tableName.getCtl(), this.getClass().getSimpleName());
+
+        if (!Env.getCurrentEnv().getAccessManager()
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/analysis/ShowColumnStmt.java:110]: Better to use this method `public boolean checkTblPriv(UserIdentity currentUser, String ctl, String db, String tbl, PrivPredicate wanted)` to and pass all `catalog`, `database` and `table` info

Comment (Nitin-Kashyap) [reply fe/fe-core/src/main/java/org/apache/doris/analysis/ShowColumnStmt.java:110]: fixed

**Review by morningman — COMMENTED** [review]
Hi @Nitin-Kashyap , I am doing another feature to support information_schema database in external catalog.
That will support this feature, see #28919

**Pull request closed without merging**